### ISSUE TITLE
[V3][Android] Blank screen on push (RN 0.60.5)

### ISSUE THREAD

**Issue by Stalder** [wix/react-native-navigation#5397]
### Issue Description

In my original project attempt to push next screen on Android leads to blank screen. When i tried to create another project from scratch to reproduce this issue I've collided with another, but similar issue. First time push works fine, but after first reload push leads to the white screen. After second reload there is just a white screen, first screen doesn't appear at all.

### Steps to Reproduce / Code Snippets / Screenshots

```sh
git clone https://github.com/Stalder/rnn-android-issue.git
cd rnn-android-issue
yarn
react-native start
react-native run-android
 ```

Navigation will fire automatically. Then shake phone and reload. And then once again.

---
### Environment
* React Native Navigation version: 3.1
* React Native version: 0.60.5
* Platform: Android
* Device info: Honor 8 lite, Android 8, debug mode

**Comment by caranico** [comment]
I encounter the same anomaly, with the same configuration.
This anomaly is not present on simulator, only on devices

**Comment by Stalder** [comment]
@caranico i've just tested both apps with Huawei Nova 3i, Android 9, and it worked nicely. Probably it depends on Android version. Which simulator did you use, internal in Android Studio, or Genymotion (or any other emulator with VM) ?

**Comment by tientruongvan1995** [comment]
I'm using LG V10, Android 7 and having the same issue
React Native Navigation version: 3.1
React Native version: 0.60.5

**Comment by Stalder** [comment]
Another device with Android 8 (Xiaomi MI 5s) works fine. Seems like it depends not on the OS version

**Comment by Kabangi** [comment]
I had a similar issue.

In my case I was upgrading from V1 to V3. I had remnant code that was showing a splash screen

i.e this line of code.

``` 
@Override
   public void addDefaultSplashLayout() {
        R.layout.splash;
  }
```

I have commented it out and everything is working as expected.

**Comment by krom-xr** [comment]
I have similar issue.
Redmi Note 3 MIUI Global 10.2
Android 6.0.1 

In some situations when push i have white screen.
When i press hardware menu button. I see apps list. I select my app again and white screen disappear

**Comment by zheldRnaskan** [comment]
I have the same issue.
Samsung galaxy S6
android 7.0
RN: 0.60.5
RNN: 3.1.1

**Comment by DaveLomber** [comment]
Having same issue on some devices.
It works perfectly with Nexus, Samsung devices.
But some users reports they see a blank screen on push (sometimes) on devices like Oppo ,Vivo, ASUS, Xiaomi  

Seems started happening after upgrade to RN60 and RNN 3.2, but not sure

**Comment by markhomoki** [comment]
I had this issue and started debugging just now. For me, updating the statusbar color on Android caused the issue. After I left the statusbar default, all screens are working properly.

**Comment by DaniShalash** [comment]
Having the same issue as well.
In my case, I have multiple themes in my app, and whenever I change the theme, I call setDefaultOptions, and then every screen/modal pushed is blank.

Same as @krom-xr 
Pressing back button until I exit the application, open it again from the apps list and everything works.

**Comment by cinder92** [comment]
@markhomoki you are right, that was the issue, thanks!

**Comment by pdziak** [comment]
I can confirm this issue and @markhomoki's solution. Will this be fixed :) ?

**Comment by trostinsky** [comment]
Also faced with this issue. @markhomoki's solution isn't work. I've not any status bar...

**Comment by herarya** [comment]
any updated?

**Comment by khorark** [comment]
> solved
> 
> ```js
>  if (Platform.Version > 25) {
>     await Navigation.setDefaultOptions({
>     ....
>     });
>   }
> ```

Isn't work for me. When I disabled set style StatusBar when I show modal it's helped, but someone I'm still catching this bug.

**Comment by trostinsky** [comment]
> > solved
> > ```js
> >  if (Platform.Version > 25) {
> >     await Navigation.setDefaultOptions({
> >     ....
> >     });
> >   }
> > ```
> 
> My version equal 26 and isn't work.
> 
> For solution need disabled:
> 
> 1. StatusBar style in options when we show modal.
>    ![image](https://user-images.githubusercontent.com/16165654/67395374-8a01d000-f5ae-11e9-802f-ab3f2629de53.png)
> 2. StatusBar style in options when we merge options.
>    ![image](https://user-images.githubusercontent.com/16165654/67395400-9ab24600-f5ae-11e9-83a9-327aa72875d4.png)
> 3. And other manipulation with status bar.
> 
> When we start app setDefaultOptions worked, even if we change statusBar style and/or color.
> If we manipulate status bar from ReactNative component it's also break RNN.

I'm not used statusBar, but have this bug when try to push new screen from opened sideMenu.
`
Navigation.push("mainStack", {
                        component: {
                            name: screen,
                            id: screen,
                            options: {
                                animations: {
                                    push: {
                                        waitForRender: true
                                    }
                                }
                            }
                        }
                    });
`

**Comment by artur-inntec** [comment]
Same issue

**Comment by arturhtml** [comment]
Same issue

**Comment by Luckfried** [comment]
I found that issue with devices which have the ability to not open apps in fullscreen.

**Comment by linhth-0720** [comment]
same issue:
 "react-native": "0.61.4",
 "react-native-navigation": "3.6.0",

**Comment by Lestoroer** [comment]
Please, fix that issue

**Comment by shaheem-khanzada** [comment]
If anybody solve this issue please help get stuck seeing blank screen

**Comment by shaheem-khanzada** [comment]
@Lestoroer i guess i found an issue remove StatusBar from all over your app even if you importing from react-native hope it will help

**Comment by Frekansapp** [comment]
removing all about statusbar working. RN version 0.61.5 RNN version 4.0.3

**Comment by glauberfc** [comment]
The solution provided by @markhomoki worked for me. Thanks, guy!

**Comment by DaniShalash** [comment]
This issue haven't been solved yet.
The lack of activity is the lack of solution, but the issue is still there.

**Comment by Stalder** [comment]
Is it possible to reopen this issue? I'm not interested already, but if issue is still not resolved, it should be open

**Comment by rodolfoviolac** [comment]
+1

**Comment by DaniShalash** [comment]
For future reference, based on @markhomoki  and @Frekansapp solutions, I was able to overcome the problem, however, it's a mix of both.

My problem was (and bare with me for obvious things):

- I have 2 themes in my app, light and dark.
- Android Status bar color and content color must change based on theme selected, and the same should apply in iOS (what's applicable of it).

Now, let's say a user is 2 screens deep in the app, and the theme changed (either by the user or following OS settings), we have 2 types of actions should be made here:

1. setDefaultOptions for future opened screens.
2. mergeOptions **OR** RN StatusBar for currently opened screens.

For point number 2, using StatusBar is what was causing the blank screen issue when setting default options, so the solution is to use merge options and remove all references of StatusBar.

A problem arrises here is that you need to keep track of the navigation history in order to use mergeOptions.
So what I did is I created a wrapper service around RNN that automatically keeps track of the navigation history along with any additional data (like passProps), which allowed me to use mergeOptions effectively (and in general provided more benefits actually).

Now, everything is working properly on Android, on iOS it's a different thing, because mergeOptions is too slow to change to statusBar content color compared to RN StatusBar.

So what I ended up doing, is using StatusBar for iOS, and mergeOptions for Android along with the RNN wrapper service, which works well on all platforms.

**Comment by NQuirmbach** [comment]
@DaniShalash could you share your code for your wrapper and describe more in detail what you've done there? I'm currently stuck with the same issue.

**Comment by guyca** [comment]
this thread completely slipped by me. Thanks https://github.com/sorokin0andrey for bringing it to my attention. If I understand the issue, changing [statusBar style](https://wix.github.io/react-native-navigation/api/options-statusBar#style) causes blank screens on Android API 27 and bellow. I've looked into this extensively in the past and wasn't able to find a solution. I suspect this is an Android issue as reported [here](https://stackoverflow.com/questions/59683122/setting-system-ui-flag-light-status-bar-makes-next-fragment-blank-and-collapsing#comment107003342_59683122).

We've faced this issue in the Wix app when changing StatusBar style dynamically with mergeOptions. In our case, changing StatusBar style wasn't actually required. But generally, we don't change styles dynamically with `mergeOptions` and `setDefaultOptions`.

If you're required to change theme, we recommend you call `setDefaultOptions` following by `setRoot`. This will render your applications root with the new theme and ensures consistent behavior across both platforms and OS versions.

**Comment by Natteke** [comment]
My problem was stretched topBar on old xiaomi.
Try this:
`Navigation.setDefaultOptions({ topBar: { visible: false, height: 0 } });`

**Comment by bozaigao** [comment]
I have the same issue, solve by remove all <StatuStatusBar/> and  Navigation.mergeOptions(this.props.componentId,{
            statusBar: {
       backgroundColor:'light'
            }
        });

**Comment by lindesvard** [comment]
We have exactly the same issue. Before the statusbar fix this was happning on all androids prior to version api 27. But after we made that change we still see problem for Xiaomi phones. As some people suggest, it feels like its not just the api level. Because we have some xiaomi devices that works perfectly fine. So super hard to reproduce/debug this.

Changes so far. 
- No statusbar components
- No mergeOptions for statusbar 
- No defaultOptions for statusbar
- Basically never touch status bar.

Version:
```
react-native-navigation: 7.8.3
react-native: 0.63.4
```
